twilio/rest/__init__.py

```diff
@@ -133,6 +137,38 @@ def request(self, method, uri, params=None, data=None, headers=None, auth=None,
             allow_redirects=allow_redirects
         )
 
+    def get_hostname(self, uri):
+        """
+        Determines the proper hostname given edge and region preferences
+        via client configuration or uri.
+
+        :param str uri: Fully qualified url
+
+        :returns: The final uri used to make the request
+        :rtype: str
+        """
+        parsed_url = urlparse(uri)
```

Comment:
nit: `if not self.edge and not self.region: return uri`

Reply:
🤦 nevermind

Reply:
eh I moved it inside anyway, mostly because i didn't see this comment, but partly because I prefer the check being inside the function with the rest of the edge/region logic.